readthedocs/rtd_tests/tests/test_notifications.py

```diff
@@ -131,3 +132,96 @@ class TestNotification(Notification):
+        backend.send(n)
+        # No email is sent for non persistent messages
+        send_email.assert_not_called()
+        self.assertEqual(PersistentMessage.objects.count(), 1)
+        self.assertEqual(PersistentMessage.objects.filter(read=False).count(), 1)
+
+        self.client.force_login(user)
+        response = self.client.get('/')
+        self.assertContains(response, 'Test success message')
+        self.assertEqual(PersistentMessage.objects.count(), 1)
+        self.assertEqual(PersistentMessage.objects.filter(read=True).count(), 1)
+
+        response = self.client.get('/')
+        self.assertNotContains(response, 'Test success message')
+
+
+@override_settings(PRODUCTION_DOMAIN='readthedocs.org')
+class SiteNotificationTests(TestCase):
+
+    class TestSiteNotification(SiteNotification):
+        name = 'foo'
+        success_message = 'simple success message'
+        failure_message = {
+            1: 'simple failure message',
+            2: '{{ object.name }} object name',
+            'three': '{{ object.name }} and {{ other.name }} render',
+        }
+        success_level = INFO_NON_PERSISTENT
+        failure_level = WARNING_NON_PERSISTENT
+
+    def setUp(self):
+        self.user = fixture.get(User)
+        self.context = {'other': {'name': 'other name'}}
+        self.n = self.TestSiteNotification(
+            self.user,
+            True,
+            context_object={'name': 'object name'},
+            extra_context=self.context,
+        )
+
+    def test_context_data(self):
+        context = {
+            'object': {'name': 'object name'},
+            'request': None,
+            'production_uri': 'https://readthedocs.org',
+            'other': {'name': 'other name'},
+        }
+        self.assertEqual(self.n.get_context_data(), context)
+
+    def test_message_level(self):
+        self.n.success = True
+        self.assertEqual(self.n.get_message_level(), INFO_NON_PERSISTENT)
+
+        self.n.success = False
+        self.assertEqual(self.n.get_message_level(), WARNING_NON_PERSISTENT)
+
+    def test_message(self):
+        self.n.reason = 1
+        self.assertEqual(self.n.get_message(True), 'simple success message')
+        self.n.reason = 'three'
+        self.assertEqual(self.n.get_message(True), 'simple success message')
+
+        self.n.reason = 1
+        self.assertEqual(self.n.get_message(False), 'simple failure message')
+        self.n.reason = 2
+        self.assertEqual(self.n.get_message(False), 'object name object name')
+        self.n.reason = 'three'
+        self.assertEqual(self.n.get_message(False), 'object name and other name render')
+
+        # Invalid reason
+        self.n.reason = None
+        with mock.patch('readthedocs.notifications.notification.log') as mock_log:
+            self.assertEqual(self.n.get_message(False), '')
+            mock_log.error.assert_called_once()
+
+
+class NonPersistentStorageTests(TestCase):
+    pass
```

Comment:
Looks like we test a lot of what we wanted to test in the above test case? maybe rename or remove this testcase.

Reply:
Meh, I followed the desing of the test classes and then I realized that the test of the backend was best in the `NotificationBackendTests`.

I will delete these lines :/